.circleci/config.yml

```diff
@@ -7,8 +17,182 @@ orbs:
+           sudo bash -c 'echo /tmp/core.%t.%e.%p >/proc/sys/kernel/core_pattern'
+
+           mkdir -p  $PWD/tmp_docker 
+      - run:
+         name: Docker Build taraxa-node
+         command: |
+           docker build -t ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} .
+      - run:
+         name: Docker Build taraxa-node-ctest
+         command: |
+          docker build -t ${IMAGE}-ctest:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} \
+            --target build .
+      - run:
+         name: Run Ctest
+         command: |
+           docker run  --rm -v $PWD/tmp_docker:/tmp \
+                       --name taraxa-node-ctest-${DOCKER_BRANCH_TAG} \
+                       ${IMAGE}-ctest:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM}\
+                        sh -c \
+                       'cd cmake-docker-build-debug/tests \
+                        && ctest --output-on-failure' 
+      - run:
+          name: Chmod core dump if fail
+          command: |
+            if [ -f tmp_docker/core* ];then
+              sudo chmod 777 tmp_docker/core*
+            fi
+          when: on_fail
+      - run: 
+         name: Run SmokeTest
+         command: |
+          if [ ! -z "$(docker network list --format '{{.Name}}' | grep -o smoke-test-net-${DOCKER_BRANCH_TAG})" ]; then
+            docker network rm \
+              smoke-test-net-${DOCKER_BRANCH_TAG} >/dev/null;
+          fi         
+          docker network create --driver=bridge \
+                      smoke-test-net-${DOCKER_BRANCH_TAG}
+          docker run --rm -d \
+              --name taraxa-node-smoke-test-${DOCKER_BRANCH_TAG} \
+              --net smoke-test-net-${DOCKER_BRANCH_TAG} \
+              ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} \
+              single
+          mkdir -p  $PWD/test_build-d/
+          sleep 30
+          http_code=$(docker run --rm  -v $PWD/test_build-d:/data \
+                --net smoke-test-net-${DOCKER_BRANCH_TAG} \
+                byrnedo/alpine-curl \
+                -sS --fail -w '%{http_code}' -o /data/http.out \
+                --url taraxa-node-smoke-test-${DOCKER_BRANCH_TAG}:7777 \
+                -d '{
+                        "jsonrpc": "2.0",
+                        "id":"0",
+                        "method": "send_coin_transaction",
+                        "params":[{
+                            "nonce": 0,
+                            "value": 0,
+                            "gas": 0,
+                            "gas_price": 0,
+                            "receiver": "973ecb1c08c8eb5a7eaa0d3fd3aab7924f2838b0",
+                            "secret": "3800b2875669d9b2053c1aff9224ecfdc411423aac5b5a73d7a45ced1c3b9dcd"
+                        }]
+                    }')
+          cat $PWD/test_build-d/http.out || true
+          if [[ $http_code -eq 200 ]] ; then
+              exit 0
+          else
+              exit $http_code
+          fi
+      - run:
+         name: Cleanup smoke tests
```

Comment:
We should run this step also if the step above fails
I don't know how to do this or even if it's possible.

Reply:
`when: always`